src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetLatestDiagnosticsForSpan.cs

```diff
@@ -107,18 +109,137 @@ private class LatestDiagnosticsForSpanGetter
                 }
             }
 
+            private async Task<bool> TryGetSyntaxAndSemanticDiagnosticsAsync(StateSet stateSet, VersionStamp textVersion, VersionStamp syntaxVersion, VersionStamp semanticVersion)
+            {
+                // unfortunately, we need to special case compiler diagnostic analyzer so that 
+                // we can do span based analysis even though we implemented it as semantic model analysis
```

Comment:
@ManishJayaswal @JohnHamby @tmeschter Probably it is worth putting more thought into this scenario.

Consider adding a "RegisterTextSpanAction" or some such, by which analyzer can state that it can compute diagnostics for a given span in a document. Obviously seems non-useful for the compiler host, but hosts like the IDE which may want to compute diagnostics for a span of a tree can do this in a more performant way if the analyzer is capable of executing and producing diagnostics on a sub-span of a tree. Our compiler diagnostic analyzer is a classic case here, and because of a lack of such an analyzer API, @heejaechang has to put this workaround and we have to use a heuristic in IDE for incremental method body analysis while typing.